src/Workleap.DomainEventPropagation.Publishing.Tests/DomainEventSchemaCacheTests.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Azure.Messaging;
+using Azure.Messaging.EventGrid;
+
+namespace Workleap.DomainEventPropagation.Publishing.Tests;
+
+public class DomainEventSchemaCacheTests
+{
+    [Fact]
+    public void Given_DomainEvent_When_RetrievingSchema_Then_ExpectedSchemaIsRetrieved()
```

Comment:
Your test should be split in two rather than test two different use cases into a single test.